Mathlib/RingTheory/UniqueFactorizationDomain.lean

```diff
@@ -1024,6 +1024,22 @@ theorem count_normalizedFactors_eq' {p x : R} (hp : p = 0 ∨ Irreducible p) (hn
   · exact count_normalizedFactors_eq hp hnorm hle hlt
 #align unique_factorization_monoid.count_normalized_factors_eq' UniqueFactorizationMonoid.count_normalizedFactors_eq'
 
+
+theorem max_powerFactor {a₀ : R} (h : a₀ ≠ 0) (hx : Irreducible x) [Nontrivial R] :
+  ∃ n : ℕ, ∃ a : R, ¬ x ∣ a ∧ a₀ = x ^ n * a := by
+  let n := (normalizedFactors a₀).count (normalize x)
```

Comment:
the formatting here doesn't match the usual lean4 style; it's best to wait for the mathlib3 PR to be merged (so that people don't accidentally merge this first, and so that they don't diverge) and the mathlib3port output to be ready (so that all you have to do is copy/paste) before making these follow-up PRs.

Reply:
Here's the mathlib3port output: https://leanprover-community.github.io/mathlib-port-status/file/ring_theory/unique_factorization_domain

Reply:
Thanks @eric-wieser and @YaelDillies. What is the best way to proceed now? I have seen online the `mathlib3` port output, should I copy-paste each modified line by hand or is there an automatic fast-forward of the modifications? It is my first time trying to modify a `mathlib4`file after an update of `mathlib3` and would appreciate some guidance. Thanks!

Reply:
I personally just copy-paste each block of modifed lines and remove the front `+`.

Reply:
The problem is that the file I see at https://leanprover-community.github.io/mathlib-port-status/file/ring_theory/unique_factorization_domain in red (which I would assume to be the old one) has nothing to do with the current state of the art of the file under my eyes. Just to understand, has the `port` automatically run? What is supposed to happen under the rug when an already ported `mathlib3` file is modified?

Reply:
These are the changes you should copy-paste:
![image](https://user-images.githubusercontent.com/14090593/236857201-81d020df-4147-4063-b4e8-d0f2c796522e.png)

Note that it includes the mathport notes, but you can just ignore them.

Reply:
What exactly is sticking you?

Reply:
Well, lines 1035 to 1041 are disgusting and I am unable to golf them to something reasonable, but was unable to find useful doc. Thanks!

Reply:
Can't you just keep the disgusting mathport output? Treat this like a porting PR; the goal is to make the content from mathlib3 compile with minimal changes, not to golf things!